### DISCUSSION TITLE
Thresholding values within input files for MODE

### DISCUSSION THREAD

**Discussion by jcorner1** [dtcenter/METplus#1994] [Configuration] [answered]
Hello, I have tried a few different methods of trying to threshold the reflectivity values at 25, 30, 35, 40 dbz. I tried doing this outside of MET but it seemed to mess up the file format and the lack of metadata messed with METs ability to regrid the obs file. Therefore, I am asking if there is a way to do it within MET itself. I will send my .conf file. I have tried a few different threshold inputs but I have not seen anything that looks how I expect it to look. Thanks for any help!
[MODE_HRRR_FCST_RADAR_OBS.conf (2).txt](https://github.com/dtcenter/METplus/files/10256004/MODE_HRRR_FCST_RADAR_OBS.conf.2.txt)

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hello @jcorner1 I see you have some questions about configuring the MODE tool via METplus. You're wondering in particular how to specify the object definition threshold.

Please take a look at the METplus [MODE: Basic Use Case](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/MODE/MODE.html#mode-basic-use-case) documentation. And for additional details about MODE itself, please see the [MODE Chapter](https://met.readthedocs.io/en/latest/Users_Guide/mode.html#mode-tool) of the MET user's guide.

Objects are defined in MODE by specifying the convolution radius and threshold. The convolution radius ([MODE_CONV_RADIUS](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-MODE_CONV_RADIUS)) determines the amount of smoothing that should applied to the raw input data, and is specified in grid units. The higher the value, the more smoothing, while setting it to 0 disables the smoothing step entirely. The convolution threshold ([MODE_CONV_THRESH](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-MODE_CONV_THRESH)) is applied to the smoothed data to define objects. 

Specifying MODE_CONV_RADIUS and MODE_CONV_THRESH will apply the settings to both the forecast and observation data. Or you can specify them separately for each field using FCST_MODE_CONV_RADIUS, FCST_MODE_CONV_THRESH, OBS_MODE_CONV_RADIUS, and OBS_MOD_CONV_THRESH.

You could specify multiple radii and thresholds, but while you're still learning about it, I'd recommend choosing a single one. Once you have reasonable output, try increasing/decreasing the radius and threshold, rerun, and observe the effect on the resolved objects.

Here's what I see in the example you sent:
```
MODE_CONV_RADIUS = 4
MODE_CONV_THRESH = >=0.5
```

So this is smoothing the data using a radius of 4. The means that the value of each raw input grid point is replaced by the mean of the values of the grid points within a circle of radius 4 around that point. The resulting values are thresholded >=0.5.

For reflectivity data with values as high as 65 dBZ or so, I would imagine that threshold would likely give you one VERY big object including most or all grid points in your domain.

I wonder if perhaps you're confused by the "CENSOR" options?
```
MODE_FCST_CENSOR_THRESH = >0&&<0.75
MODE_FCST_CENSOR_VAL = 40
```
This is applied to the RAW input data, as it is being read. I'd recommend NOT using the data censoring options, unless you have a good reason to do so. Here's an excerpt from [this section](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=censor_thresh#fcst) of the MET user's guide describing how it works:

The “censor_thresh” entry is an array of thresholds to be applied to the input data. The “censor_val” entry is an array of numbers and must be the same length as “censor_thresh”. These arguments must appear together in the correct format (threshold and number). For each censor threshold, any input values meeting the threshold criteria will be reset to the corresponding censor value. An empty list indicates that no censoring should be performed. The censoring logic is applied prior to any regridding but after the convert function. All statistics are computed on the censored data. These entries may be used to apply quality control logic by resetting data outside of an expected range to the bad data value of -9999. These entries are not indicated in the metadata of any output files, but the user can set the “desc” entry accordingly.

**Reply by jcorner1** [reply]
That did it! I was a bit confused on what created a threshold on the raw data and the documentation confused me a bit.

**Reply by JohnHalleyGotway** [reply]
Great, thanks for confirming! I'll go ahead and lock this discussion to prevent future posts, which are easily overlooked. If more questions arise, feel free to create a new discussion topic.
